### PR TITLE
[THREESCALE-7084] 3scale API will not be authenticated by the user_cookie

### PR DESCRIPTION
**Pull request by jsmadis**
**What this PR does / why we need it**:

HTTP request to the 3scale API won't be authenticated via user_session inside cookies.

Main changes:

- Analytics (frontend) endpoint is moved from path `/stats/...` to `/stats/api/...`
- New controllers for the analytics were created. (Copy of the REST API ones).
- REST API endpoints are no longer authenticated with session/cookie.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-7084

### CODE REVIEW

**Comment by thomasmaas** [comment]
This change means all users will have to manually authenticale calls by pasting personal token in api docs?

**Comment by jsmadis** [comment]
@thomasmaas yes, your assumption is correct.

File: doc/active_docs/Analytics API.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "apis": [
     {
-      "path": "/stats/applications/{application_id}/usage.{format}",
+      "path": "/admin/api/stats/applications/{application_id}/usage.{format}",
```
Comment (hallelujah) [inline doc/active_docs/Analytics API.json:4]: This is a breaking change!

I am not so sure we want that

File: app/javascript/src/Stats/buyer/stats_application.jsx [hunk 2]
```diff
@@ -43,13 +43,13 @@ export class StatsApplicationChartManager extends StatsUsageChartManager {
   }
 
   _metricUrl (id) {
-    return `/stats/applications/${id}/summary.json?version=2.0`
+    return `/stats/api/applications/${id}/summary.json?version=2.0`
```
Comment (hallelujah) [inline app/javascript/src/Stats/buyer/stats_application.jsx:46]: So this is a temporary measure. we would need to clean that by properly announcing a deprecation notice
On the API side

`stats/applications` => `admin/api/stats/applications`

I wonder if we could create the routes already

File: config/routes.rb [hunk 3]
```diff
@@ -444,6 +444,29 @@ def call(env)
 
   # These are API routes, beware
   namespace :stats do
+    namespace :api do
+      # horrible copy paste, but we need to create separate endpoints
```
Comment (hallelujah) [inline config/routes.rb:448]: Especially it is not a "real" API but a JS endpoint

**Review by hallelujah — APPROVED** [review]

**Pull request merged**